tensorbay/utility/file.py

```diff
@@ -19,6 +19,55 @@
 from tensorbay.utility.repr import ReprMixin
 
 
+class Url:
+    """Url is a class used to get and update the url.
+
+    Arguments:
+        url: The url.
+        updater: A function used to update the url.
+
+    """
+
+    def __init__(self, url: str, updater: Callable[..., Optional[str]]) -> None:
+        self._url = url
+        self._updater = updater
+        self._getter: Callable[..., str] = lambda: self._url
+
+    def update(self) -> None:
+        """Update the url."""
+        url = self._updater()
```

Comment:
the updater is called without argument, which means its type should be `Callable[[], Optional[str]]`